zrml/prediction-markets/src/lib.rs

```diff
@@ -2527,49 +2550,116 @@ mod pallet {
             Ok(report.outcome.clone())
         }
 
+        /// Handle a market resolution, which is currently in the disputed state.
+        /// Returns the resolved outcome of a market.
         fn resolve_disputed_market(
             market_id: &MarketIdOf<T>,
             market: &MarketOf<T>,
-        ) -> Result<OutcomeReport, DispatchError> {
+        ) -> Result<(OutcomeReport, Weight), DispatchError> {
```

Comment:
I think we should define a new type for returning a result and the weight:
```rust
#[derive(Debug)]
struct ResultWithWeight<T> {
    result: T,
    weight: Weight
}
```

Reply:
85131915f06664f8c2d7857e8e6809760fca86a7